pkg/kubernetes/nulecule.js

```diff
@@ -0,0 +1,263 @@
+            var dfd = $.Deferred();
+            debug("writing to " + install_dir + "/answers.conf");
+            var file = cockpit.file(install_dir + "/answers.conf");
+            var req = file.replace(JSON.stringify(data))
+                .done(function(){
+                    dfd.resolve();
+                    file.close();
+                })
+                .fail(function(ex){
+                    console.warn(ex);
+                    dfd.reject(ex);
+                    file.close();
+                });
+
+            var promise = dfd.promise();
+            promise.cancel = function cancel() {
+                req.cancel();
+                file.close();
+            };
+
+            return promise;
+        }
+
+        function installrun(ptype, image) {
+            var deferred = $.Deferred();
+            var promise;
+            var process;
+
+            if (!install_dir) {
+                deferred.reject();
+                return deferred.promise();
+            }
+
+            if(ptype === "install") {
+                //atomic install <IMAGE> command internally calls atomicapp install <IMAGE>
+                process = cockpit.spawn(["/usr/bin/atomic", "install", image],{ err: "out", superuser: true , directory: install_dir, pty: true });
+                debug("Installing image: " + image + " in folder " + install_dir);
+                deferred.notify(_("Installing ..."));
+            } else {
+                //atomic run --opt2=--answers="/tmp/answers.conf" <IMAGE>
+                //internally calls...
+                //atomicapp --answers=/tmp/answers.conf run <IMAGE>
+                // Note : OPT2 has to be in this format for atomic command to work :--answers="/tmp/answers.conf" , 
+                //i.e "--opt2=--answers=\"" + install_dir + "/answers.conf\" "
+                process = cockpit.spawn(["/usr/bin/atomic", "run", "--opt2=--answers=\"" + install_dir + "/answers.conf\" ", image],{ err: "out", superuser: true , directory: install_dir , pty: true});
+                debug("Running image: " + image + " in folder " + install_dir);
+                deferred.notify(_("Running ..."));
+            }
+
+            process.always(function() {
+                    process.close();
+                })
+                .stream(function(text) {
+                    text = String(text);
+                    debug("buffer="+text);
+                    var msgl;
+                    if(text.indexOf("atomicapp.status.info.message") > -1) {
+                        msgl = text.split('atomicapp.status.info.message=');
+                        if(msgl.length > 1)
+                            deferred.notify(msgl[1]);
+
+                    } else if(text.indexOf("atomicapp.status.answer.message") > -1) {
+                        msgl = text.split('atomicapp.status.answer.message=');
+                        if(msgl.length > 1) {
+                            self.answers = JSON.parse(msgl[1]);
```

Comment:
Got some JSON parse errors here when running this. Do we have any way of fixing the server side code so we get valid json?